src/libpmemobj/heap.c

```diff
@@ -928,6 +938,21 @@ heap_end(struct palloc_heap *h)
 	return &last_zone->chunks[last_zone->header.size_idx];
 }
 
+/*
+ * heap_arena_create -- create a new arena and push it to the vector
+ */
+int
+heap_arena_create(struct palloc_heap *heap)
+{
+	struct heap_rt *h = heap->rt;
+	if (VEC_PUSH_BACK(&h->arenas, heap_arena_new()))
+		return -1;
+
+	h->narenas = (unsigned)VEC_SIZE(&h->arenas);
```

Comment:
The `narenas` variable is used for scheduling for threads to arenas. The purpose of arena create is to manually manage scheduling.

These two things should be separate. As a next step, simply create a WRITE handler for heap.narenas which will set it to the user-provided value (which is less or equal VEC_SIZE(&h->arenas)).